File: AtlasWeb/Core/Modules/Tibbi_Surec/Yogun_Bakim_Modulu/BallardNeuromuscularScoreFormViewModel.ts
//$29C8492B
import { BaseViewModel } from "NebulaClient/Model/BaseViewModel";
import { BallardNeuromuscular } from "NebulaClient/Model/AtlasClientModel";
import { Type } from "../../../wwwroot/app/NebulaClient/ClassTransformer";

export class BallardNeuromuscularScoreFormViewModel extends BaseViewModel {
    @Type(() => BallardNeuromuscular)
    public _BallardNeuromuscular: BallardNeuromuscular = new BallardNeuromuscular();
}

File: AtlasWeb/Core/Modules/Tibbi_Surec/Yogun_Bakim_Modulu/BallardNeuromuscularScoreFormViewModel.cs
//$84E400DA
using System;
using System.Linq;
using Core.Models;
using Infrastructure.Filters;
using Infrastructure.Models;
using TTInstanceManagement;
using TTObjectClasses;
using TTDefinitionManagement;
using TTUtils;

namespace Core.Controllers
{
    public partial class BallardNeuromuscularServiceController
    {
        partial void PreScript_BallardNeuromuscularScoreForm(BallardNeuromuscularScoreFormViewModel viewModel, BallardNeuromuscular ballardNeuromuscular, TTObjectContext objectContext)
        {
            EpisodeAction episodeAction = ballardNeuromuscular.EpisodeAction;
            if (episodeAction == null)
            {
                Guid? selectedEpisodeActionObjectID = viewModel.GetSelectedEpisodeActionID();
                if (selectedEpisodeActionObjectID.HasValue)
                {
                    episodeAction = objectContext.GetObject<EpisodeAction>(selectedEpisodeActionObjectID.Value);
                    ballardNeuromuscular.EpisodeAction = episodeAction;
                }
            }
            ballardNeuromuscular.EntryUser = Common.CurrentResource;
        }
    }
}

namespace Core.Models
{
    public partial class BallardNeuromuscularScoreFormViewModel: BaseViewModel
    {
    }
}
